tox.ini

```diff
@@ -0,0 +1,88 @@
+[tox]
+envlist =
+    check-{style,security,build}
+    test{,-warnings,-cov}-xdist
+    test-numpy{120,121,122}
+    test-{jwst,romancal}-xdist
+    build-{docs,dist}
+
+# tox environments are constructed with so-called 'factors' (or terms)
+# separated by hyphens, e.g. test-devdeps-cov. Lines below starting with factor:
+# will only take effect if that factor is included in the environment name. To
+# see a list of example environments that can be run, along with a description,
+# run:
+#
+#     tox -l -v
+#
+
+[testenv:check-style]
+description = check code style, e.g. with ruff
+skip_install = true
+deps =
+    ruff
+commands =
+    ruff . {posargs}
+
+[testenv:check-security]
+description = run bandit to check security compliance
+skip_install = true
+deps =
+    bandit>=1.7
+commands =
+    bandit -r -ll src
+
+[testenv]
+description =
+    run tests
+    jwst: of JWST pipeline
+    romancal: of Romancal pipeline
+    oldestdeps: with the oldest supported version of key dependencies
+    warnings: treating warnings as errors
+    cov: with coverage
+    xdist: using parallel processing
+extras =
+    test
+deps =
+    xdist: pytest-xdist
+    cov: pytest-cov
+    jwst: jwst[test] @ git+https://github.com/spacetelescope/jwst.git
+    romancal: romancal[test] @ git+https://github.com/spacetelescope/romancal.git
+    oldestdeps: minimum_dependencies
+pass_env =
+    CRDS_*
+    CI
+set_env =
+    jwst: CRDS_SERVER_URL=https://jwst-crds.stsci.edu
+    romancal: CRDS_SERVER_URL=https://roman-crds.stsci.edu
+package = 
+    !cov: wheel
+    cov: editable
+allowlist_externals =
+    echo
+commands_pre =
+    oldestdeps: minimum_dependencies stpipe --filename requirements-min.txt
+    oldestdeps: pip install -r requirements-min.txt
+    pip freeze
+commands =
+    pytest \
+    warnings: -W error \
+    xdist: -n auto \
+    jwst: --pyargs jwst --ignore-glob=timeconversion --ignore-glob=associations \
+    romancal: --pyargs romancal \
+    cov: --cov=src/stpipe --cov-config=pyproject.toml --cov-report=term-missing --cov-report=xml \
```

Comment:
the `*/scripts/*` in `jwst` are ignored for testing purposes by `jwst`. In stdatamodels, I had to add https://github.com/spacetelescope/stdatamodels/blob/514ff63dc4f357c197d1b7649b30817373e4d5a9/tox.ini#L54 in order to avoid the failures in these tests.